fabtests/component/dmabuf-rdma/fi-rdmabw-rocr.c

```diff
@@ -0,0 +1,1282 @@
+
+	while ((c = getopt(argc, argv, "2b:d:D:e:p:m:M:n:t:gPB:rRsS:x:hv")) != -1) {
+		switch (c) {
+		case '2':
+			bidir = 1;
+			break;
+		case 'b':
+			batch = atoi(optarg);
+			if (batch <= 0) {
+				fprintf(stderr,
+					"Batch too small, adjusted to 1\n");
+				batch = 1;
+			} else if (batch > TX_DEPTH) {
+				fprintf(stderr,
+					"Batch too large, adjusted to %d\n",
+					TX_DEPTH);
+				batch = TX_DEPTH;
+			}
+			break;
+		case 'd':
+			gpu_dev_nums = strdup(optarg);
+			break;
+		case 'D':
+			domain_names = strdup(optarg);
+			break;
+		case 'e':
+			if (strcasecmp(optarg, "rdm") == 0)
+				ep_type = FI_EP_RDM;
+			else if (strcasecmp(optarg, "msg") == 0)
+				ep_type = FI_EP_MSG;
+			break;
+		case 'p':
+			prov_name = strdup(optarg);
+			break;
+		case 'm':
+			parse_buf_location(optarg, &loc1, &loc2, MALLOC);
+			break;
+		case 'n':
+			iters = atoi(optarg);
+			break;
+		case 't':
+			if (strcasecmp(optarg, "read") == 0)
+				test_type = READ;
+			else if (strcasecmp(optarg, "write") == 0)
+				test_type = WRITE;
+			else if (strcasecmp(optarg, "send") == 0)
+				test_type = SEND;
+			break;
+		case 'P':
+			use_proxy = 1;
+			break;
+		case 'B':
+			proxy_block = atoi(optarg);
+			if (proxy_block < MIN_PROXY_BLOCK) {
+				fprintf(stderr,
+					"Block size too small, adjusted to %d\n",
+					MIN_PROXY_BLOCK);
+				proxy_block = MIN_PROXY_BLOCK;
+			}
+			break;
+		case 'r':
+			reverse = 1;
+			break;
+		case 'R':
+			use_dmabuf_reg = 1;
+			break;
+		case 's':
+			use_sync_ofi = 1;
+			break;
+		case 'S':
+			msg_size = parse_size(optarg);
+			break;
+		case 'M':
+			max_size = parse_size(optarg);
+			proxy_block = max_size;
+			break;
+		case 'x':
+			prepost = atoi(optarg);
+			break;
+		case 'v':
+			verify = 1;
+			break;
+		default:
+			usage(argv[0]);
+			exit(-1);
+			break;
+		}
+	}
+
+	if (msg_size > 0 && msg_size > max_size) {
+		max_size = msg_size;
+		proxy_block = msg_size;
+		fprintf(stderr,
+			"Max_size smaller than message size, adjusted to %zd\n",
+			max_size);
+	}
+
+	if (max_size * batch / batch != max_size) {
+		fprintf(stderr,
+			"Buffer_size = Max_size (%zd) * Batch (%d) overflows\n",
+			max_size, batch);
+		exit(-1);
+	}
+
+	if (argc > optind) {
+		client = 1;
+		server_name = strdup(argv[optind]);
+	}
+
+	s = getenv("PMI_RANK");
+	if (s) {
+		rank = atoi(s);
+		client = rank % 2;
+		port += rank >> 1;
+		if (!client && server_name) {
+			free(server_name);
+			server_name = NULL;
+		}
+	}
+
+	buf_location = client ? loc2 : loc1;
+
+	sockfd = connect_tcp(server_name, port);
+	if (sockfd < 0) {
+		fprintf(stderr, "Cannot create socket connection\n");
+		exit(-1);
+	}
+
+	initiator = (!reverse && server_name) || (reverse && !server_name);
+
+	if (use_dmabuf_reg)
+		dmabuf_reg_open();
+
+	// multi-GPU test doesn't make sense if buffers are on the host 
+	enable_multi_gpu = buf_location != MALLOC && buf_location != HOST;
+    num_gpus = rocr_init(gpu_dev_nums, enable_multi_gpu); //num_gpus=1;
```

Comment:
It would be good if `rocr_init` has the same signature as `xe_init`.

Reply:
you mean the header? Isn't current parameter the same as `xe_init`